src/main/kotlin/no/nav/syfo/dialogmote/database/MotesvarKafkaSentAtQueries.kt

```diff
@@ -0,0 +1,76 @@
+package no.nav.syfo.dialogmote.database
+
+import no.nav.syfo.application.database.toList
+import no.nav.syfo.dialogmote.api.domain.Dialogmotesvar
+import no.nav.syfo.dialogmote.api.domain.SenderType
+import no.nav.syfo.dialogmote.domain.DialogmoteSvarType
+import no.nav.syfo.domain.PersonIdentNumber
+import java.sql.Connection
+import java.sql.ResultSet
+import java.util.*
+
+const val queryGetArbeidstakeresUnpublishedMotesvar =
+    """
+        SELECT m.uuid as mote_uuid, ma.personident, v.svar_type,  v.uuid as varsel_uuid
+        FROM motedeltaker_arbeidstaker_varsel v
+        LEFT JOIN motedeltaker_arbeidstaker ma ON v.motedeltaker_arbeidstaker_id = ma.id
+        LEFT JOIN mote m ON ma.mote_id = m.id
+        WHERE v.svar_tidspunkt IS NOT NULL 
+        AND svar_published_to_kafka_at IS NULL;
+    """
+
+fun Connection.getArbeidstakeresUnpublishedMotesvar(): List<Dialogmotesvar> {
+    return this.prepareStatement(queryGetArbeidstakeresUnpublishedMotesvar).use {
+        it.executeQuery().toList { toDialogmotesvar(SenderType.ARBEIDSTAKER) }
+    }
+}
+
+const val queryGetArbeidsgiveresUnpublishedMotesvar =
+    """
+        SELECT m.uuid as mote_uuid, a.personident, v.svar_type,  v.uuid as varsel_uuid
+        FROM motedeltaker_arbeidsgiver_varsel v
+        LEFT JOIN motedeltaker_arbeidsgiver ma ON v.motedeltaker_arbeidsgiver_id = ma.id
+        LEFT JOIN mote m ON ma.mote_id = m.id
+        LEFT JOIN motedeltaker_arbeidstaker a ON m.id = a.mote_id
+        where v.svar_tidspunkt IS NOT NULL
+        AND svar_published_to_kafka_at IS NULL; 
+    """
+
+fun Connection.getArbeidsgiveresUnpublishedMotesvar(): List<Dialogmotesvar> {
+    return this.prepareStatement(queryGetArbeidsgiveresUnpublishedMotesvar).use {
+        it.executeQuery().toList { toDialogmotesvar(SenderType.ARBEIDSGIVER) }
+    }
+}
+
+const val queryGetBehandleresUnpublishedMotesvar =
+    """
+        SELECT m.uuid as mote_uuid, ma.personident, s.svar_type, s.uuid as varsel_uuid
+        FROM motedeltaker_behandler_varsel_svar s
+        LEFT JOIN motedeltaker_behandler_varsel mbv ON s.motedeltaker_behandler_varsel_id = mbv.id
+        LEFT JOIN motedeltaker_behandler mb ON mbv.motedeltaker_behandler_id = mb.id
+        LEFT JOIN mote m ON mb.mote_id = m.id
+        LEFT JOIN motedeltaker_arbeidstaker ma ON m.id = ma.mote_id 
```

Comment:
🤠  https://genius.com/Beyonce-irreplaceable-lyrics

Reply:
Men ja, litt vanskelig å se for seg hvordan dette blir uten å snakke gjennom det. Men en `LEFT JOIN` er vel det samme som en `LEFT OUTER JOIN`?

Reply:
Ja, nå ser jeg Geir kommenterte på Inner join lenger oppe.

Reply:
Ja, left join er shorhand for left outer join og bør bare brukes der man faktisk ønsker å få ut resultater der høyre side av join'en mangler.